include/cantera/thermo/MixtureFugacityTP.h

```diff
@@ -1,577 +1,572 @@
+
+    //@}
+    //! @name Initialization Methods - For Internal use
+    /*!
+     * The following methods are used in the process of constructing
+     * the phase and setting its parameters from a specification in an
+     * input file. They are not normally used in application programs.
+     * To see how they are used, see importPhase().
+     */
+    //@{
+
+    virtual bool addSpecies(shared_ptr<Species> spec);
+    virtual void setStateFromXML(const XML_Node& state);
+
+protected:
+    //! @name Special Functions for fugacity classes
+    //! @{
+
+    //!  Calculate the value of z
+    /*!
+     *  \f[
+     *        z = \frac{P v}{R T}
+     *  \f]
+     *
+     *  returns the value of z
+     */
+    doublereal z() const;
+
+    //! Calculate the deviation terms for the total entropy of the mixture from
+    //! the ideal gas mixture
+    /*
+     * Here we use the current state conditions
+     *
+     * @returns the change in entropy in units of J kmol-1 K-1.
+     */
+    virtual doublereal sresid() const;
+
+    //! Calculate the deviation terms for the total enthalpy of the mixture from
+    //! the ideal gas mixture
+    /*
+     * Here we use the current state conditions
+     *
+     * @returns the change in entropy in units of J kmol-1.
+     */
+    virtual doublereal hresid() const;
+
+    //! Estimate for the saturation pressure
+    /*!
+     * Note: this is only used as a starting guess for later routines that
+     * actually calculate an accurate value for the saturation pressure.
+     *
+     * @param TKelvin  temperature in kelvin
+     * @return the estimated saturation pressure at the given temperature
+     */
+    virtual doublereal psatEst(doublereal TKelvin) const;
+
+public:
+    //! Estimate for the molar volume of the liquid
+    /*!
+     * Note: this is only used as a starting guess for later routines that
+     * actually calculate an accurate value for the liquid molar volume. This
+     * routine doesn't change the state of the system.
+     *
+     * @param TKelvin  temperature in kelvin
+     * @param pres     Pressure in Pa. This is used as an initial guess. If the
+     *                 routine needs to change the pressure to find a stable
+     *                 liquid state, the new pressure is returned in this
+     *                 variable.
+     * @returns the estimate of the liquid volume. If the liquid can't be
+     *          found, this routine returns -1.
+     */
+    virtual doublereal liquidVolEst(doublereal TKelvin, doublereal& pres) const;
+
+    //! Calculates the density given the temperature and the pressure and a
+    //! guess at the density.
+    /*!
+     * Note, below T_c, this is a multivalued function. We do not cross the
+     * vapor dome in this. This is protected because it is called during
+     * setState_TP() routines. Infinite loops would result if it were not
+     * protected.
+     *
+     *  -> why is this not const?

```

Comment:
This looks like a TODO comment, was this here before? Can this be resolved?

Reply:
This comment is old, I don't know who wrote it. Also it is unclear what should exactly be constant (variable/function..). I have removed it now.